### ISSUE TITLE
Question regarding the direction of `memory` in GEM implementation (`new - old`)

### ISSUE THREAD

**Issue by iknki** [shaoshuo-ss/FedTracker#3]
Hi authors,

Thank you for sharing your excellent work and the source code! I am currently reproducing your paper for my research in federated learning and it has been incredibly helpful.

While studying the implementation of the Gradient Episodic Memory (GEM) module, I noticed a very interesting detail regarding the construction of the `memory` in the `load_global_model` function, and I was hoping to get your insights on it.

In the code, the memory is updated using the weight differences:
```python
self.memory[key] = torch.sub(new_weights, old_weights).to(device)
```
Mathematically, according to standard gradient descent ($W_{new} = W_{old} - \eta \cdot g$), the difference $W_{new} - W_{old}$ is essentially proportional to the **negative** past gradient ($-g_{past}$).

In the standard GEM algorithm, the projection is triggered when the current gradient conflicts with the past gradient, i.e., when $\langle g_{current}, g_{past} \rangle < 0$. 
However, since the `memory` here stores the negative direction, the condition `dotp < 0` in your training loop effectively becomes:
$\langle g_{current}, -g_{past} \rangle < 0 \implies \langle g_{current}, g_{past} \rangle > 0$.

This seems to imply that the GEM projection is triggered when the current task's gradient and the main task's gradient are **in the same direction** (rather than conflicting). 

I was wondering if this reversed direction was a deliberate design choice specifically tailored for the watermarking/fine-tuning scenario in federated learning? For instance, does this act more like a regularization mechanism pulling the model back towards $W_{old}$? 

I would really appreciate it if you could share the intuition behind this specific calculation. Thanks again for your fantastic contribution!
